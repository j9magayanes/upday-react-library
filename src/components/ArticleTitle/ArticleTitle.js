import React from 'react';
import './articleTitle.css';
import 'upday-web-components/components/articleTitle';

export const ArticleTitle = (
    {title, className, id }
) => {
  return (
   <article-title  className={className}>{title}</article-title>
  );
};