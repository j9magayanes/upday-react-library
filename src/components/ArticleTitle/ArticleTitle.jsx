import React, { Children } from 'react';
import PropTypes from 'prop-types';
import './articleTitle.css';
import { argsHash } from '@storybook/blocks'

export const ArticleTitle = (
    { primary, backgroundColor, size, label, title,  ...props }
) => {
  return (
   <h1 id={props.id} className={['storybook-title', `storybook-title--${size}`].join(' ')}>{title}</h1>
  );
};