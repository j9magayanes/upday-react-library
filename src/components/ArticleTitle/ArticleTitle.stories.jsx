import React, { Children } from 'react';
import PropTypes from 'prop-types';
import './articleTitle.css';
import { argsHash } from '@storybook/blocks';
import { ArticleTitle } from './ArticleTitle';

export default {
    title: 'Form/ArticleTitle',
    component: ArticleTitle,
  };
  
  const Template = (args) =><ArticleTitle   {...args} />;
  
  export const Meinestadt = Template.bind({});
  Meinestadt.args = {
    label: 'meinestadt',
    title: 'meinestadt'
  };
  
  export const Web = Template.bind({});
  Web.args = {
    title: 'Web',
    size: 'web'
  };
  
 