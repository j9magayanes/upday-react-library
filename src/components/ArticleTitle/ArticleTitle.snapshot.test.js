import React from 'react';
import renderer from 'react-test-renderer';
import { ArticleTitle } from './ArticleTitle';

test('renders ArticleTitle component correctly', () => {
  const props = {
    title: 'Your Title',
    className: 'your-class-name',
    id: 'your-id',
  };

  const component = renderer.create(<ArticleTitle {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
