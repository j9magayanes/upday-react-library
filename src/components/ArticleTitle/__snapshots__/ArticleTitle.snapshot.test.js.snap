// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticleTitle component correctly 1`] = `
<article-title
  className="your-class-name"
>
  Your Title
</article-title>
`;
